core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -318,6 +318,7 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                             continue;
                         }
                         try {
+                            mapperService.checkIfAddMappingAllowed(entry.getKey());
```

Comment:
You cannot do the check here. You have no way to know if the type is created automatically by an index operation or through a legitimate create index.